### PR TITLE
Refactor pack install

### PR DESCRIPTION
**Pull request by chaws**
In this PR:
1. Ensure `public` pack installations **DO NOT** save a copy of the pdsc file in `.Local/` (bug spot by @jkrech)
2. Ensure `public` installed packs **ALWAYS** have a copy of the pdsc file in `.Web/` and that `non-public` packs have a copy of the pdsc file in `.Local/`
2.1. Ensure that all packs installed have the `<CMSIS_PACK_ROOT>/Vendor/PackName/x.y.z/Vendor.PackName.pdsc` file
2.2. Propagate the same behavior for packs installed via direct file or via url
3. Change the order of URL retrieval for a pack. The order now is `packPdsc.releaseTag.URL`, and if that is empty build the url using `packPdsc.URL/Vendor/PackName.pdsc`
3.1. If installing a pack via `cpackget pack add Vendor.PackName.1.0.0` and the releaseTag does not exist, raise `"pack version not found in the pdsc file"`
4. Avoid updating `.Web/index.pidx` automatically on `pack add`. Instead, let the user know the public index might be outdated and show how to update it.
5. By default `cpackget pack add Vendor.PackName` will attempt to read `.Web/Vendor.PackName.pdsc`. 
5.1. If the file is not present, then look for this tag in `.Web/index.pidx`.
5.1.1. If the tag **IS** present
5.1.1.1. Attempt to download the pdsc using the tag's URL attribute + `Vendor.PackName.pdsc`, and place it in `.Web/Vendor.PackName.pdsc`
5.1.1.2. If the download of the pdsc file fails, raise `"the URL for the pack pdsc file seems not to exist or it didn't return the file"`
5.1.2. If the tag is **NOT** present
5.1.2.1. Attempt to read `.Local/Vendor.PackName.pdsc`
5.1.2.2. If file exists, retrieve pack's URL as mentioned in step `3`
5.1.2.3. If file does not exist, raise `"URL for the pack cannot be determined. Please consider updating the public index: cpackget index https://keil.com/pack/index.pidx"`

It still needs a bit polishing and other small features that need some attention after these changes. I'll update this PR accordingly.

### CODE REVIEW

File: cmd/installer/pack.go [hunk 1]
```diff
@@ -27,8 +27,8 @@ import (
 type PackType struct {
 	xml.PdscTag
 
-	// isPublic tells whether the pack exists in the public index or not
-	isPublic bool
+	// IsPublic tells whether the pack exists in the public index or not
+	IsPublic bool
```
Comment (jkrech) [inline cmd/installer/pack.go:31]: Why did you decide to capitalize IsPublic but isInstalled begins with a lower case. What is the coding guidelines you follow?

Comment (chaws) [reply cmd/installer/pack.go:31]: Originally that was not intended to be exported, but it turned out to be necessary to check in tests. I was trying to export only the necessary fields, but eventually I think I'll converge to all variables there being exported (capitalized first letter).

Comment (jkrech) [reply cmd/installer/pack.go:31]: Ah, sorry for my lack of knowledge of the GO language. Thanks for explaining.

**Comment by chaws** [comment]
@jkrech I'm merging this now and should be landing another PR soon early next week with another set of changes. I'll break the PRs into smaller ones

**Pull request merged**